syntax = "proto3";

package databaseinstance;

service DatabaseInstanceService {
    rpc Read(ReadRequest) returns (ReadResponse);
    rpc Write(WriteRequest) returns (WriteResponse);
}

message ReadRequest {
    string id = 1;
}

message ReadResponse {
    int32 stockValue = 1;
}

message WriteRequest {
    string id = 1;
    int32 stockValue = 2;
    repeated string hosts = 3;
    repeated string ports = 4;
}

message WriteResponse {
  bool isSuccess = 1;
  string failedhost = 2;
  string failedport = 3;
}